packages/compass-e2e-tests/tests/logging.test.ts

```diff
@@ -340,7 +342,8 @@ describe('Logging and Telemetry integration', function () {
         // resulting in duplicate test names.
         it(`logs "${expected.msg}" (${i})`, function () {
           const actualLogIndex = criticalPathActualLogs.findIndex(
-            ({ id }, index) => id === expected.id && !testedIndexes.has(index)
+            ({ id }, index) =>
+              id === mongoLogId(expected.id) && !testedIndexes.has(index)
```

Comment:
I don’t think the first condition can ever be true … maybe we need to modify the types a bit, or use `{ id }: any` here

Reply:
It complained that it could never work because one's a mongo id and the other's a number :)

Reply:
Right, but this now creates a new object and compares reference equality of that new object to another object, which is never true in JS

Reply:
(Plus, `id` actually *is* a number in the actuaĺ logs after processing)

Reply:
is this correct, then? https://github.com/mongodb-js/mongodb-log-writer/blob/2422aab82eb64fca45d8278ee99f84bb726916c2/src/index.ts#L34

Reply:
It is correct for the writing side, yes.